From 550dfab63f14994687dc520040d149cd04ecc454 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david.llewellyn-jones@jolla.com>
Date: Wed, 25 Mar 2020 11:59:52 +0200
Subject: [PATCH] Set qmf-accountscheck to be run by the booster

In order for qmf-accountscheck to gain the privileges needed to read
the accounts database, it must be invoked by the booster (it also needs
a .privileges file, added elsewhere).

This change sets the service file to use the booster.
---
 src/tools/systemd/messageserver5-accounts-check.service | 4 +++-
 src/tools/tools.pro                                     | 2 ++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/tools/systemd/messageserver5-accounts-check.service b/src/tools/systemd/messageserver5-accounts-check.service
index c2ec29ea..f5194daa 100644
--- a/src/tools/systemd/messageserver5-accounts-check.service
+++ b/src/tools/systemd/messageserver5-accounts-check.service
@@ -1,12 +1,14 @@
 [Unit]
 Description=check accounts for messageserver5
+Requires=booster-qt5.service
 After=oneshot-user.service
 After=jolla-settings-accounts-update.service
 Before=messageserver5.service
+After=booster-qt5.service
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/qmf-accountscheck
+ExecStart=/usr/bin/invoker -o --type=generic --global-syms /usr/bin/qmf-accountscheck
 
 [Install]
 WantedBy=user-session.target
diff --git a/src/tools/tools.pro b/src/tools/tools.pro
index 689ad320..15aa630e 100644
--- a/src/tools/tools.pro
+++ b/src/tools/tools.pro
@@ -4,3 +4,5 @@ SUBDIRS = messageserver
 contains(DEFINES,USE_ACCOUNTS_QT) {
     SUBDIRS += accountscheck
 }
+
+OTHER_FILES += systemd/*.service
-- 
2.26.2

